tls/s2n_x509_validator.h

```diff
@@ -62,6 +62,32 @@ struct s2n_x509_validator {
     int state;
 };
 
+struct s2n_x509_cert {
+    X509 *cert;
+};
```

Comment:
How are applications expected to get access to this struct? I'm worried about confusing it with our existing certificate opaque structure: https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L2440 Should they be the same structure? If not, won't it be confusing that all our of APIs to examine certs don't work on both structures?

Reply:
I didn't consider this. Since the only operation on the certificate that's needed in the callback is getting the issuer hash, I think it makes sense to have this be a getter on the CRL lookup context rather than returning a whole struct from that context. Since this context struct is in the next PR, I'll wait to add this function there.